.circleci/config.yml

```diff
@@ -1,29 +1,17 @@
-version: 2
+version: 2.1
 jobs:
   build:
     docker:
-      - image: circleci/python:3.7.0
+      - image: cimg/python:3.12
     steps:
       - checkout
       - setup_remote_docker
       - run:
           name: Build container
-          command: docker build -t ds-python .
+          command: docker build --target test -t ds-python .
       - run:
           name: Verify build completed
           command: docker run ds-python /bin/bash -c "echo BUILDS OK"
       - run:
-          name: Check that scipy links to OpenBLAS
-          command: docker run ds-python python -c "from scipy.linalg import _fblas"
-      - run:
-          name: Check that numpy imports from bash shell
-          command: docker run -t ds-python /bin/bash -c "python -c 'import numpy'"
-      - run:
-          name: Verify that numpy does not link to MKL
-          command: docker run ds-python python -c "from numpy.distutils import system_info; assert system_info.get_info('mkl') == {}"
```

Comment:
`numpy.distutils` has been removed for Python 3.12. This check was added likely for the previous conda set-up, but we aren't using conda anymore.